netplugin/netd.go

```diff
@@ -464,28 +465,36 @@ func handleStateEvents(netPlugin *plugin.NetPlugin, crt *crt.CRT, opts cliOpts,
 	go handleEndpointEvents(netPlugin, crt, opts, retErr)
 }
 
-func handleEvents(netPlugin *plugin.NetPlugin, crt *crt.CRT, opts cliOpts) error {
-
-	recvErr := make(chan error, 1)
+func handleEvents(netPlugin *plugin.NetPlugin, crt *crt.CRT, opts cliOpts) {
+	recvErr := make(chan error)
 
 	//monitor and process state change events
 	go handleStateEvents(netPlugin, crt, opts, recvErr)
+	startDockerEventPoll(netPlugin, crt, recvErr, opts)
+
+	go func() {
+		for {
+			err := <-recvErr
```

Comment:
I think this will cause `startDockerEventPoll()` to be called even if the error was set by `handleStateEvents()` in which case we should probably exit. May be we can create a separate error channel for `startDockerEventPoll()` to signal errors?

Also even when `startDockerEventPoll()` errors out may be we shall check for a specific error (like `broken pipe` or something more specific that can indicate a `docker` daemon restart) and continue to exit in other scenarios.